### PR TITLE
Optimize how nydus meta snapshot is prepared and enable errcheck linter

### PR DESCRIPTION
**Pull request by changweige**
1. Reduce unnecessary database queries
2. enable `errcheck` linter

This PR is very likely to reduce snapshot setup latency.

### CODE REVIEW

File: pkg/backend/oss.go [hunk 1]
```diff
@@ -141,7 +142,9 @@ func (b *OSSBackend) push(ra content.ReaderAt, blobDigest digest.Digest) error {
 		}
 
 		if err := g.Wait(); err != nil {
-			b.bucket.AbortMultipartUpload(imur)
+			if err := b.bucket.AbortMultipartUpload(imur); err != nil {
+				log.L.WithError(err)
```
Comment (imeoer) [inline pkg/backend/oss.go:146]: Does this print message to stderr? We'd better not do this in a public package.

File: pkg/blob/blob_manager.go [hunk 2]
```diff
@@ -146,7 +146,11 @@ func (b *Manager) PrepareBlobLayer(snapshot storage.Snapshot, labels map[string]
 	if err != nil {
 		return errors.Wrap(err, "rename temp file as blob file")
 	}
-	os.Chmod(blobFile.Name(), 0440)
+
+	err = os.Chmod(blobPath, 0440)
+	if err != nil {
+		return err
```
Comment (imeoer) [inline pkg/blob/blob_manager.go:152]: Better to wrap it.

File: pkg/backend/oss.go (hunk 1 above)
Comment (changweige) [reply pkg/backend/oss.go:146]: No. It prints to logger

Comment (changweige) [reply pkg/backend/oss.go]: fixed

File: pkg/blob/blob_manager.go (hunk 2 above)
Comment (changweige) [reply pkg/blob/blob_manager.go]: fixed

**Review by imeoer — APPROVED** [review]

**Pull request merged**